slime_environments/environments/SlimeEnvMultiAgent.py

```diff
@@ -296,7 +309,15 @@ def _wrap(self, x: int, y: int):
 
     # learners act
     def step(self, action: int):
-        agent_in_charge = self.agent_selection  # ID of agent
+        if(self.terminations[self.agent_selection] or self.truncations[self.agent_selection]):
+            self._was_dead_step(action)
+            return
+        
+        agent_in_charge = self.agent_name_mapping[self.agent_selection]  # ID of agent
+        
+        self.process_agent(agent_in_charge) #
+        self.state[agent_in_charge] = action #can ignore this

```

Comment:
could you explain what this is for?

Reply:
As I mentioned in the description of PR:
> Renamed last() method to process_agent() method. Reason: last() method from the parent class AECEnv can be used directly. The logic inside former last() method can be used inside the step() method, just before calculating the rewards thus preserving the flow of logic in previous environment (Earlier: last() -> step(). Now: Inside step() - logic from previous last() and the previous step() logic)

I just replaced your last() method logic to process_agent() and made the current last() method compatible to the definition of pettingZoo's. 

But, if you meant about the self.state (line 319), I just referred from the documentation page ([here](https://pettingzoo.farama.org/content/environment_creation/#:~:text=%23%20stores%20action%20of%20current%20agent)).

Reply:
yes I was referring only to the last line, don't know why also others were included in the comment (my bad possibly).
From the linked documentation its purpose is unclear, but let's leave it as it could impact hidden processes inside PettingZoo.
Thanks, nice catch!

Reply:
for self.state, it might be a mandatory variable for pettingZoo API, but need to check. But, it doesn't impact our code, since I have not used that variable anywhere, other than just initializing.

Reply:
it seems unused also in the code snippet linked to pettingzoo, that's why I said that we can just leave it there to avoid risky breaking something on pettingzoo side